--- hyde.scm.orig	2020-10-21 22:57:56.795903959 +0900
+++ hyde.scm	2020-10-26 15:57:16.739658557 +0900
@@ -358 +358,2 @@
-  (let ((output-file (make-pathname (output-dir) (pathname-relative-from (source-dir) path))))
+  (let* ((output-file (make-pathname (output-dir) (pathname-relative-from (source-dir) path)))
+         (output-file (irregex-replace "(.*)/([0-9]{4})-([0-9]{2})-([0-9]{2})-(.*)" output-file 1 "/" 2 "/" 3 "/" 4 "/" 5)))
@@ -417,0 +419,4 @@
+(define (default-page-vars-for-date page)
+  (let ((m (irregex-search "([0-9]{4}-[0-9]{2}-[0-9]{2})-.+\\.md$" (page-source-path page))))
+    (if m `((date . ,(irregex-match-substring m 1))) `())))
+
@@ -437 +442,2 @@
-                  (page (update-page page vars: (append local-page-vars
+                      (page (update-page page vars: (append (default-page-vars-for-date page)
+                                                            local-page-vars
@@ -446 +452,4 @@
-                            (with-output-to-file (make-output-path path page)
+                                (let ((rel-path (make-output-path path page)))
+                                  (create-directory (string-append (current-directory) "/" (pathname-directory rel-path))
+                                                    #t)
+                                  (with-output-to-file rel-path
@@ -451 +460 @@
-                                            (lambda () (wrap-with-layouts (reader)))))))))))
+                                                  (lambda () (wrap-with-layouts (reader))))))))))))
